### PR TITLE
Lint string

### PR DESCRIPTION
**Pull request by SimenB**
See #161 

I created a V2 branch, so we can make breaking changes in there, then merge into develop when we think it's done. Mostly to allow `develop` to still receive other changes.

Contains changes from #273 as well, will rebase once that's merged

### CODE REVIEW

**Comment by SimenB** [comment]
I'll get to this after #366 gets merged

**Comment by SimenB** [comment]
Rebased now. Need to rethink skipping exit, as `done` now has the responsibility of formatting errors for reporters. should probably move out of `done`.

**Comment by SimenB** [comment]
Alright, this one is _finally_ ready for review!

/cc @wojciechczerniak

**Comment by SimenB** [comment]
Started in July 2015. About time, I'd say...

File: src/core/lintString.js [hunk 1]
```diff
@@ -0,0 +1,33 @@
+'use strict'
+
+/**
+ * @description wrapper for parse for programmatic use
+ * @param {String} string the text to parse
+ * @param {Object} [config] an object containing config to lint by
+ * @param {String} [filename] the name of the file
+ * @returns {Object} an object containing the complete report from linting
+ */
+var lintString = function( string, config, filename ) {
```
Comment (wojciechczerniak) [inline src/core/lintString.js]: Where's the connection between lintString and a file name? I thought that idea behind this API is to use it without file.

**Comment by SimenB** [comment]
Yeah, but you might want a filename in the report.

Say the gulp plugin, it has a file, but I won't have to do IO, as it's already read. But it still corresponds to a file.

Eslint also takes a (optional) filename: http://eslint.org/docs/developer-guide/nodejs-api#executeontext

File: src/core/lintString.js (hunk 1 above)
Comment (SimenB) [inline src/core/lintString.js]: Should have answered inline, cut&paste:

Yeah, but you might want a filename in the report.

Say the gulp plugin, it has a file, but Stylint won't have to do IO, as it's already read. But it still corresponds to a file.

Eslint also takes a (optional) filename: http://eslint.org/docs/developer-guide/nodejs-api#executeontext

**Pull request merged**